dockerfiles/Dockerfile-build-crypt

```diff
@@ -0,0 +1,145 @@
+    && sed -i '/^CXXFLAGS/ s/$/ -D__STDC_NO_ATOMICS__=1/' Makefile \
+    && emmake make -j 2 all \
+    && emmake make install
+
+# Build libcrypt4gh
+FROM emscripten/emsdk:latest AS LIBCRYPT4GH
+
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/include/ /emsdk/upstream/include/
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/lib/ /emsdk/upstream/lib/
+
+RUN sudo apt-get update \
+    && sudo apt-get upgrade -y \
+    && sudo apt-get install -y autoconf
+
+RUN git clone https://github.com/silverdaz/libcrypt4gh.git
+
+# We'll skip linking libraries since emcc only produces static libraries
+# Linking sodium at this point causes a linker conflict – thus cutting out $(LIBS)
+RUN  export EMCC_CFLAGS="-I/emsdk/upstream/include -L/emsdk/upstream/lib" \
+    && export LDFLAGS="-L/emsdk/upstream/lib" \
+    && cd libcrypt4gh \
+    && autoreconf \
+    && sed -i 's/$(LIBS) //' Makefile.in \
+    && emconfigure ./configure --prefix=/emsdk/upstream \
+    && emmake make \
+    && emmake make install
+
+# Build libcrypt4gh-keys
+FROM emscripten/emsdk:latest AS LIBCRYPT4GHKEYS
+
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/include/ /emsdk/upstream/include/
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/lib/ /emsdk/upstream/lib/
+
+COPY --from=OPENSSL /emsdk/upstream/include/ /emsdk/upstream/include/
+COPY --from=OPENSSL /emsdk/upstream/lib/ /emsdk/upstream/lib/
+
+RUN sudo apt-get update \
+    && sudo apt-get upgrade -y \
+    && sudo apt-get install -y autoconf build-essential
+
+RUN git clone https://github.com/silverdaz/libcrypt4gh-keys.git
+
+# We'll skip linking libraries since emcc only produces static libraries
+# Linking sodium at this point causes a linker conflict – thus cutting out $(LIBS)
+RUN export EMCC_CFLAGS="-I/emsdk/upstream/include -L/emsdk/upstream/lib" \
+    && export LDFLAGS="-L/emsdk/upstream/lib" \
+    && cd libcrypt4gh-keys \
+    && autoreconf \
+    && sed -i 's/$(LIBS) //' Makefile.in \
+    && emconfigure ./configure --prefix=/emsdk/upstream --with-openssl=/emsdk/upstream \
+    && emmake make \
+    && emmake make install
+
+# Build wasm encryption
+FROM emscripten/emsdk:latest AS WASMCRYPT
+
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/include/ /emsdk/upstream/include/
+COPY --from=SODIUM /src/libsodium-stable/libsodium-js-sumo/lib/ /emsdk/upstream/lib/
+
+COPY --from=LIBCRYPT4GH /emsdk/upstream/include/ /emsdk/upstream/include/
+COPY --from=LIBCRYPT4GH /emsdk/upstream/lib/ /emsdk/upstream/lib/
+
+COPY --from=LIBCRYPT4GHKEYS /emsdk/upstream/include/ /emsdk/upstream/include/
+COPY --from=LIBCRYPT4GHKEYS /emsdk/upstream/lib/ /emsdk/upstream/lib/
+
+COPY swift_browser_ui_frontend/wasm/ /src/
+
+RUN export EMCC_CFLAGS="-I/emsdk/upstream/include -L/emsdk/upstream/lib" \
+    && export EMCC_FORCE_STDLIBS=libc \
+    && emmake make
+
+# Build UI with encryption
+FROM node:14.17.1-alpine3.12 as FRONTEND
```

Comment:
this needs updating

Reply:
They just merged alpine 3.15, but we currently have 3.14 in other images: `node:14.18.2-alpine3.14`